src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/PubSubController.cs

```diff
@@ -0,0 +1,142 @@
+        /// </summary>
+        public IEnumerable<string> Strings;
+    }
+
+    /// <summary>
+    ///     Publishing and subscribing to messages on a topic.
+    /// </summary>
+    public class PubSubController : IpfsController
+    {
+        /// <summary>
+        ///     Creates a new controller.
+        /// </summary>
+        public PubSubController(IDfsService dfs) : base(dfs) { }
+
+        /// <summary>
+        ///     List all the subscribed topics.
+        /// </summary>
+        [HttpGet] [HttpPost] [Route("pubsub/ls")]
+        public async Task<PubsubTopicsDto> List()
+        {
+            return new PubsubTopicsDto
+            {
+                Strings = await IpfsCore.PubSubApi.SubscribedTopicsAsync(Cancel)
+            };
+        }
+
+        /// <summary>
+        ///     List all the peers associated with the topic.
+        /// </summary>
+        /// <param name="arg">
+        ///     The topic name or null/empty for "all topics".
+        /// </param>
+        [HttpGet] [HttpPost] [Route("pubsub/peers")]
+        public async Task<PubsubPeersDto> Peers(string arg)
+        {
+            var topic = string.IsNullOrEmpty(arg) ? null : arg;
+            var peers = await IpfsCore.PubSubApi.PeersAsync(topic, Cancel);
+            return new PubsubPeersDto
+            {
+                Strings = peers.Select(p => p.Id.ToString())
+            };
+        }
+
+        /// <summary>
+        ///     Publish a message to a topic.
+        /// </summary>
+        /// <param name="arg">
+        ///     The first arg is the topic name and second is the message.
+        /// </param>
+        [HttpGet] [HttpPost] [Route("pubsub/pub")]
+        public async Task Publish(string[] arg)
+        {
+            if (arg.Length != 2)
+                throw new ArgumentException("Missing topic and/or message.");
+            var message = arg[1].Select(c => (byte) c).ToArray();
+            await IpfsCore.PubSubApi.PublishAsync(arg[0], message, Cancel);
+        }
+
+        /// <summary>
+        ///     Subscribe to messages on the topic.
+        /// </summary>
+        /// <param name="arg">
+        ///     The topic name.
+        /// </param>
+        [HttpGet] [HttpPost] [Route("pubsub/sub")]
+        public async Task Subscribe(string arg)
+        {
+            await IpfsCore.PubSubApi.SubscribeAsync(arg, message =>
+            {
+                // Send the published message to the caller.
+                var dto = new MessageDto(message);
+                StreamJson(dto);
+            }, Cancel);
+
+            // Send 200 OK to caller; but do not close the stream
+            // so that published messages can be sent.
+            Response.ContentType = "application/json";
+            Response.StatusCode = 200;
+            await Response.Body.FlushAsync();
+
+            // Wait for the caller to cancel.
+            try
+            {
+                await Task.Delay(-1, Cancel);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)